components/kserve/kserve.go

```diff
@@ -34,13 +32,28 @@ var (
 // Verifies that Kserve implements ComponentInterface.
 var _ components.ComponentInterface = (*Kserve)(nil)
 
+// +kubebuilder:validation:Pattern=`^(Serverless|RawDeployment)$`
+type DefaultDeploymentMode string
+
+var (
+	// Serverless will be used as the default deployment mode for Kserve. This requires Serverless and ServiceMesh operators configured as dependencies.
+	Serverless DefaultDeploymentMode = "Serverless"
+	// RawDeployment will be used as the default deployment mode for Kserve.
+	RawDeployment DefaultDeploymentMode = "RawDeployment"
+)
+
 // Kserve struct holds the configuration for the Kserve component.
 // +kubebuilder:object:generate=true
 type Kserve struct {
 	components.Component `json:""`
 	// Serving configures the KNative-Serving stack used for model serving. A Service
 	// Mesh (Istio) is prerequisite, since it is used as networking layer.
 	Serving infrav1.ServingSpec `json:"serving,omitempty"`
+	// Configures the default deployment mode for Kserve. This can be set to 'Serverless' or 'RawDeployment'.
+	// The value specified in this field will be used to set the default deployment mode in the 'inferenceservice-config' configmap for Kserve
+	// +kubebuilder:validation:Enum=Serverless;RawDeployment
+	// +kubebuilder:default=Serverless
```

Comment:
IMO, the default should be empty to:
* be able to have a minimal DSC (i.e. reduce the verbosity);
* and, if empty, this would let the operator choose the best value, given the `managementState` of Serverless (because `managementState` fields don't allow empties).

Reply:
@danielezonca @zdtsw @bartoszmajsak wdyt about this approach?

Reply:
I agree to avoid "polluting" to much the API so fine for me to have an empty default with the meaning of "Serverless"

Reply:
So the reason we are not using the pointers in general for fields in DSCI/DSC is there was a previous requirement that if a field is not set, it should still be visible for users with a default value, so that they know readily what configurations are available. [Ref](https://github.com/opendatahub-io/opendatahub-operator/issues/324)

However, this is a field within a component spec, so I think it should be okay to set it as pointer.

Reply:
Removed the `// +kubebuilder:default=Serverless` line to make the default empty and added logic to infer `Serverless` as the default deployment mode if a value is not explicitly provided 
cc @israel-hdez

Reply:
should we add some description into README for this new feature  (even it wont break nightly)?

Reply:
> should we add some description into README for this new feature (even it wont break nightly)?

We, probably, should have a small conversation about where ODH community docs should live. Right now, I'm writing something for KServe authorization and I'm planning to put it on opendatahub-io/kserve repo. I'd cover how to configure the feature (the operator part) and also how to use the feature. Clearly, the how to configure may also belong to this repo. But, then, docs would be scattered.

I'm :+1: on having community docs, but I'm :-1: on not having a single place to host them. At the very least, we should have a single entry point and, then, links (and cross-links) between the relevant repos.